ONSdigital/census-rh-service
# Defence in depth and least privilege minimum acceptable security changes

# Motivation and Context
As part of defence in depth, which is the minimum expected security standard for government projects, application accounts should have _least privilege_ so that they cannot be used maliciously, if an application is compromised: we cannot make every account a 'god like' super user... that undermines all security efforts.

# What has changed
Removed RH's attempts to create queues, exchanges, bindings, which require full admin privileges: all those queues, exchanges and bindings will pre-exist, because RM has a very strict and tightly security controlled process, to roll out infrastructure like that.

# How to test?
It should work like-for-like, provided the necessary queues etc. exist, as they already do in WL, BL, GL, PreProd & Prod.

If you want to test locally, you should create a testing script to create whatever queues you need for testing. RM codebases have many examples of how you can do this.

# Links
Trello: I'll put a link to a trello ticket if and when exists, but this just tech debt: closing a security loophole.
